obspy/clients/filesystem/tsindex.py

```diff
@@ -0,0 +1,1820 @@
+            nslc_list.append(nslc)
+        nslc_list.sort()
+        return nslc_list
+
+    def get_availability_extent(self, network=None, station=None,
+                                location=None, channel=None, starttime=None,
+                                endtime=None):
+        """
+        Get a list of tuples [(network, station, location, channel,
+        earliest, latest)] containing data extent info for time series
+        included in the tsindex database.
+
+        :type network: str
+        :param network: Network code of requested data (e.g. "IU").
+            Wildcards '*' and '?' are supported.
+        :type station: str
+        :param station: Station code of requested data (e.g. "ANMO").
+            Wildcards '*' and '?' are supported.
+        :type location: str
+        :param location: Location code of requested data (e.g. "").
+            Wildcards '*' and '?' are supported.
+        :type channel: str
+        :param channel: Channel code of requested data (e.g. "HHZ").
+            Wildcards '*' and '?' are supported.
+        :type starttime: :class:`~obspy.core.utcdatetime.UTCDateTime`
+        :param starttime: Start of requested time window.
+        :type endtime: :class:`~obspy.core.utcdatetime.UTCDateTime`
+        :param endtime: End of requested time window.
+        :rtype: list(tuple(str, str, str, str,
+            :class:`~obspy.core.utcdatetime.UTCDateTime`,
+            :class:`~obspy.core.utcdatetime.UTCDateTime`))
+        :returns: A list of tuples [(network, station, location, channel,
+            earliest, latest)...] containing data extent info for time series
+            included in the tsindex database.
+        """
+        summary_rows = self._get_summary_rows(network, station,
+                                              location, channel,
+                                              starttime, endtime)
+
+        availability_extents = []
+        for row in summary_rows:
+            extent = (row.network, row.station, row.location, row.channel,
+                      UTCDateTime(row.earliest), UTCDateTime(row.latest))
+            availability_extents.append(extent)
+        availability_extents.sort()
+        return availability_extents
+
+    def get_availability(self, network=None, station=None, location=None,
+                         channel=None, starttime=None, endtime=None,
+                         include_sample_rate=False,
+                         merge_overlap=False):
+        """
+        Get a list of tuples [(network, station, location, channel,
+        starttime, endtime),...] containing data availability info for
+        time series included in the tsindex database.
+
+        If ``include_sample_rate=True``, then a tuple containing the sample
+        rate [(net, sta, loc, cha, start, end, sample_rate),...] is returned.
+
+        If ``merge_overlap=True``, then all time spans that overlap are merged.
```

Comment:
What happens to the ``seamless`` traces (across day line, as explained above, for example) ?

Reply:
Seamless data, a series without time tears, are presented as seamless regardless of any arbitrary splitting in the index due to data file separation (e.g. day files of otherwise continuous data).

Reply:
Ok, so does this merge_overlap parameter consider those, although they don't overlap, literally?